executor/write.go

```diff
@@ -1007,9 +1022,21 @@ func (e *InsertExec) getKeysNeedCheck(rows [][]types.Datum) ([][]keyWithDupError
 // batchMarkDupRows marks rows with duplicate errors as nil.
 // All duplicate rows were marked and appended as duplicate warnings
 // to the statement context in batch.
-func (e *InsertExec) batchMarkDupRows(rows [][]types.Datum) ([][]types.Datum, error) {
+func batchMarkDupRows(i interface{}, rows [][]types.Datum) ([][]types.Datum, error) {
```

Comment:
I think pass `ctx` and `t` instead of interface{} is better.

Reply:
I used to add a limitation for this two types. Anyway fixed. PTAL @coocood